drivers/spi/spi_ll_stm32.c

```diff
@@ -449,11 +632,72 @@ static int transceive(struct device *dev,
 	return ret;
 }
 
+#ifdef CONFIG_SPI_STM32_DMA
+static int transceive_dma(struct device *dev,
+		      const struct spi_config *config,
+		      const struct spi_buf_set *tx_bufs,
+		      const struct spi_buf_set *rx_bufs,
+		      bool asynchronous, struct k_poll_signal *signal)
+{
+	const struct spi_stm32_config *cfg = DEV_CFG(dev);
+	struct spi_stm32_data *data = DEV_DATA(dev);
+	SPI_TypeDef *spi = cfg->spi;
+	int ret;
+
+	if (!tx_bufs && !rx_bufs) {
+		return 0;
+	}
+
+	if (asynchronous) {
+		return -ENOTSUP;
```

Comment:
Would be nice to support it, the non-dma part supports it already. I think that's trivial to support.

Reply:
Same as `transceive` function
  @tbursztyka  What could be the asynchronous mode with a DMA transfer ?

Reply:
It's like interrupt mode, you notify the caller through its k_poll_signal the transaction is done (or got some error).
You have set a dma_callback above that can be used for that I think.
So if asynchronous mode is requested by the user, you just don't run the loop at line 667 nor cs control nor release: you do these I believe in the dma cb.

But it's ok you could add this later in another PR.